lib/cocoapods/installer/xcode/pods_project_generator/pod_target_dependency_installer.rb

```diff
@@ -107,6 +107,43 @@ def wire_test_native_targets(pod_target, installation_result, pod_target_install
                 Project.add_cached_dependency(test_native_target, cached_dependency)
               end
             end
+
+            if app_host_target_label = pod_target.app_host_target_label(test_spec)
+              app_host_pod_target_label, app_host_target_label = *app_host_target_label
+              wire_test_native_target_app_host(test_native_target, pod_target, pod_target_installation_results, project, metadata_cache, app_host_pod_target_label, app_host_target_label)
+            end
+          end
+        end
+
+        def wire_test_native_target_app_host(test_native_target, pod_target, pod_target_installation_results, project, metadata_cache, app_host_pod_target_label, app_host_target_label)
+          if dependency_installation_result = pod_target_installation_results[app_host_pod_target_label]
+            unless app_native_target = dependency_installation_result.app_host_target_labelled(app_host_target_label)
+              raise Informative, "Did not find target with label #{app_host_target_label} in the set of targets installed for #{app_host_pod_target_label}."
+            end
+
+            dependent_test_project = app_native_target.project
+            if dependent_test_project != project
+              project.add_subproject_reference(dependent_test_project, project.dependencies_group)
+            end
+
+            app_host_target_names = app_native_target.resolved_build_setting('PRODUCT_NAME', true)
+            test_native_target.build_configurations.each do |configuration|
```

Comment:
nit: Fairly large function here. Setting up the build configs/build settings seems like a concern that can be pulled out. Non-blocking, just a thought.